packages/react-frontend/src/CreateUser.jsx

```diff
@@ -19,9 +21,76 @@ function CreateUser(props) {
       ...prevPerson,
       [name]: value,
     }));
+
+    // Validate username and password on input change
+    if (name === "name") {
+      validateUsername(value);
+    } else if (name === "password") {
+      const strength = strengthOmeter(value);
+      const message = strengthMessage(strength);
+      setPasswordError(message);
+    }
+  }
+
+  function validateUsername(username) {
+    if (username.length < 1) {
+      setUsernameError("U53RN4M3 2 5H0R7 :(");
+    } else {
+      setUsernameError(null);
+    }
+  }
+
+  function strengthOmeter(password) {
+    let strength = 0;
+
+    if (password.length >= 5) {
+      strength += 1;
+    }
+
+    if (/[A-Z]/.test(password)) {
+      strength += 1;
+    }
+
+    if (/[a-z]/.test(password)) {
+      strength += 1;
+    }
+
+    if (/\d/.test(password)) {
+      strength += 1;
+    }
+
+    if (/[^A-Za-z0-9]/.test(password)) {
+      strength += 1;
+    }
+
+    return strength;
+  }
+
+  function strengthMessage(strength) {
```

Comment:
Could you use an if-else if-else statement so that it's easier to see, instead of having to read case 0, case 1...